input/pagecontent/processing_model.md

```diff
@@ -0,0 +1,136 @@
+    2. For each Selection Structure `sel` in `S.unionAll[]`
+        * Define `u` as  `ValidateColumns(sel, Ret)`
+            * If the list of names from `u0` is different from the list of names from `u`, throw "Union Branches Inconsistent"
+            * Otherwise, continue
+    3. For each Column `col` in `u0`
+        * Append `col` to `Ret`
+
+4. Return `Ret`
+
+### Process a Resource (entry point)
+
+**Purpose:** This step emits all rows produced by a ViewDefinition on an input Resource, by setting up a recursive call.
+
+**Inputs**
+* `V`: a `ViewDefinition`
+* `R`: a FHIR Resource to process with `V`
+
+**Emits:** one output row at a time
+
+1. Ensure resource type is correct
+    * If `R.resourceType` is different from `V.resource`, return immediately without emitting any rows
+    * Otherwise, continue
+2. If `V.where` is defined, ensure constraints are met
+    * Evaluate `fhirpath(V.where.path, R)` to determine whether `R` is a candidate for `V`
+        * If `R` is not a candidate for `V`, return immediately without emitting any rows
+        * Otherwise, continue
+3. Emit all rows from `Process(S, V)`
+
+### `Process(S, N)` (recursive step)
+
+**Purpose:** This step emits all rows for a given Selection Structure and Node. We first generates sets of "partial rows" (i.e., sets of incomplete column bindings from the various clauses of `V`) and combine them to emit complete rows. For example, if there are two sets of partial rows:
+
+* `[{"a": 1},{"a": 2}]` with bindings for the variable `a`
+* `[{"b": 3},{"b": 4}]` with bindings for the variable `b`
+
+Then the Cartesian product of these sets consists of four complete rows:
+
+```js
+[
+  {"a": 1, "b": 3},
+  {"a": 1, "b": 4},
+  {"a": 2, "b": 3},
+  {"a": 2, "b": 4}
+]
+```
+
+**Inputs**
+* `S`: a Selection Structure
+* `N`: a Node (element) from a FHIR resource
+
+**Errors**
+* Multiple values found but not expected for column
+
+**Emits:** One output row at a time
+
+1. Define a list of Nodes `foci` as
+    *  If `S.forEach` is defined: `fhirpath(S.forEach, N)`
+    *  Else if `S.forEachOrNull` is defined: `fhirpath(S.forEachOrNull, N)`
+    *  Otherwise: `[N]` (a list with just the input node)
+
+2. For each element `f` of `foci`
+    1. Initialize an empty list `parts` (each element of `parts` will be a list of partial rows)
+    2. Process Columns:
+        * For each Column `col` of `S.column`, define `val` as  `fhirpath(col.path, f)`
+            1. Define `b` as a row whose column named `col.name` takes the value
+                * If `val` was the empty set: `null`
+                * Else if `val` has a single element `e`: `e`
+                * Else if `col.collection` is true: `val`
+                * Else: throw "Multiple values found but not expected for column"
+            2. Append `[b]` to `parts`
+            * (Note: append a list so the final element of `parts` is now a list containing the single row `b`).
+    3. Process Selects:

```

Comment:
I think the logic described here is correct, but maybe a clarifying comment for the case of `select` structures in a parent->child->grandchild relationship could help.  We wouldn't cartesian join all rows between {parent, child, grandchild}, but each is cartesian joined with its child, and so recursively back up the stack. 

(And of course since the `columns` in a selection structure only produce a single row, a "cartesian join" between a parent select and one child select is just to duplicate the parent column across child rows.)

Reply:
Thanks -- I've added a note to this effect.